CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+# HERMES development workflow
+
+## Preamble
+
+> **Branching is cheap!**
+>
+> Work in small packets, put up small pull requests!
+>
+> Aim for quick turnaround times!
+
+## Branching
+
+We loosely follow a mixture of [GitHubFlow](https://docs.github.com/en/get-started/quickstart/github-flow) and [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/):
+
+`main`
+    - Stable branch
+    - Merges come only from `develop` or a hotfix branch (i.e., when something needs to be fixed in "production")
+
+`develop`
+    - Unstable development branch
+    - Last "stable" is tagged: tag before attempting to break something
+
+`feature/<describe-feature>` (including issue id when exists: `feature/62-fix-broken-thing`)
+    - Branch from last tag on `develop`
+
```

Comment:
```suggestion
`feature/<describe-feature>` (including issue id when exists: `feature/62-add-fancy-feat`)
    - Branch from last tag on `develop`

`fix/<describe-fix>` (including issue id when exists: `fix/63-fix-broken-thing`)
    - Branch from last tag on `develop`
```

It felt VERY weird creating a branch with `feature` in its name to *fix* something.

Reply:
**objection**: Distinguish between "fix" and "features" seems fine to me... but what else will be there? "refactoring", "release", ... So we should not over-engineer our processes as long as we are not even successfully merging our current branches.

Reply:
I agree with not over-engineering this. The `fix/` and `feature/` prefixes are OK, but also unnecessary IHMO. Github's tag can be used to mark PRs as `fix`, `feature`, `refactoring`, `release`, .... This also allows having multiple tags, e.g. adding a `feature` may also involve `refactoring`.

Reply:
To be clear, these were introduced based on what GitFlow uses:

- `feature/` for non-hotfix work, i.e., work based on `develop`. This includes work that is semantically a `fix`, which you have to get used to (it's possible, done it for the last 3 years in a larger project).
- `hotfix/` for work against `main`, that fixes something in already released code
- `release/` for work that will go into the next release, e.g., releng, updating CHANGELOG.md, etc.
- and then `develop` and `main` (in lieu of `master`) ...

Reply:
Not that I care very much, apart from that I think we should either stick to this, or go fully free-form with a light suggestions to include issue numbers if available. Everything else is over-engineering, I agree.

Reply:
I did not know about these details of the GitFlow definition. With your explanation though, @sdruskat , the given set makes a lot of sense. There is a also not need to have `fix` in addition to `feature` (even though `feature/fix` may feel strange), because the prefix strictly describes what kind of branch it is and thus what it may be merged into, not what is on the branch.

I agree with @sdruskat  to either use those, or go free-form. Either way, we should not care what people do on their forks, so if someone opens a PR merging, e.g. `jkelling:develop` into `hermes-hmc:develop`, we should not complain.

Reply:
Fine fine fine I'll get used to it. This discussion is already kinda over engineering 😜

Let's stick with the proposal and 🚢 it so we get going 🙈